#! /bin/sh /usr/share/dpatch/dpatch-run
## 22_launch_moc-qt4.dpatch by Brian Nelson <pyro@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Ensure the Qt4 version of moc is launched.

@DPATCH@
diff -urNad qt4-x11-4.3.0~rc1~/tools/qdbus/tools/qdbuscpp2xml/qdbuscpp2xml.cpp qt4-x11-4.3.0~rc1/tools/qdbus/tools/qdbuscpp2xml/qdbuscpp2xml.cpp
--- qt4-x11-4.3.0~rc1~/tools/qdbus/tools/qdbuscpp2xml/qdbuscpp2xml.cpp	2007-05-06 17:54:54.000000000 +0200
+++ qt4-x11-4.3.0~rc1/tools/qdbus/tools/qdbuscpp2xml/qdbuscpp2xml.cpp	2007-05-18 09:44:18.000000000 +0200
@@ -376,7 +376,7 @@
         else {
             // run moc on this file
             QProcess proc;
-            proc.start(QLibraryInfo::location(QLibraryInfo::BinariesPath) + QLatin1String("/moc"), QStringList() << QFile::decodeName(argv[i]), QIODevice::ReadOnly | QIODevice::Text);
+            proc.start(QLibraryInfo::location(QLibraryInfo::BinariesPath) + QLatin1String("/moc-qt4"), QStringList() << QFile::decodeName(argv[i]), QIODevice::ReadOnly | QIODevice::Text);
 
             if (!proc.waitForStarted()) {
                 fprintf(stderr, PROGRAMNAME ": could not execute moc! Aborting.\n");
